daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/CidContainer.scala

```diff
@@ -93,6 +97,23 @@ trait CidContainer[+A] {
       case rcoid: Value.RelativeContractId => Left(rcoid)
     }(self)
 
+  // Sets the suffix of any the V1 AbsoluteContractId `coid` of the container that are not already suffixed.
+  // Uses `f(coid.discriminator)` as suffix.
+  // Fails if encounters relative contract id or a V0 contract id.
+  final def suffixCid[B](f: crypto.Hash => Bytes)(
```

Comment:
@andreaslochbihler-da @oliverse-da 
Here is the new method to suffix new Contract ids.
See the `TransactionSpec` to see how to use it on transaction. It works similarly on nodes, values ...

Reply:
Thanks for the pointer. I'll study this in more detail when I get to switch to V1 contract ids in Canton.